src/devices/Card/Icode/IcodeCard.cs

```diff
@@ -0,0 +1,355 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using Iot.Device.Common;
+using Microsoft.Extensions.Logging;
+using Iot.Device.Card.Mifare;
+
+namespace Iot.Device.Card.Icode
+{
+    /// <summary>
+    /// A Icode card class Supports ICODE SLIX,ICODE SLIX2,ICODE DNA,ICODE 3
+    /// </summary>
+    public class IcodeCard
+    {
+        private const byte BytesPerBlock = 4;
+
+        // This is the actual RFID reader
+        private readonly CardTransceiver _rfid;
+
+        private readonly ILogger _logger;
+
+        // the size of response
+        private ushort _responseSize;
+
+        /// <summary>
+        /// The tag number detected by the reader
+        /// </summary>
+        public byte Target { get; set; }
+
+        /// <summary>
+        /// The command to execute on the card
+        /// </summary>
+        public IcodeCardCommand Command { get; set; }
```

Comment:
Consider only making specific Commands public and everything else to be hidden as an implementation detail